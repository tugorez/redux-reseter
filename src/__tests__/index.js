import { reset, reseter } from '../index';

const reducer = (state = 0, { type, payload = 1 }) => {
  if (type === 'ADD') return state + payload;
  if (type === 'SUB') return state - payload;
  return state;
};

describe('reducer', () => {
  it('should compute the operations', () => {
    const action1 = { type: 'ADD' };
    const action2 = { type: 'ADD' };
    const action3 = { type: 'SUB' };
    const action4 = { type: 'ADD' };
    const actions = [action1, action2, action3, action4];
    const state = actions.reduce(reducer, 0);
    expect(state).toMatchSnapshot();
  });
});

describe('redux-reseter', () => {
  describe('reset', () => {
    it('creates a valid action', () => {
      const action = reset({ type: 'ADD' }, { type: 'SUB' }, { type: 'SUB' });
      expect(action).toMatchSnapshot();
    });
  });

  describe('reseter', () => {
    const reseted = reseter(reducer);

    it('should wait for a reset action and reset the state', () => {
      const actions = [
        { type: 'ADD' },
        { type: 'ADD' },
        { type: 'ADD' },
        reset(),
      ];
      const state = actions.reduce(reseted, undefined);
      expect(state).toMatchSnapshot();
    });
  });
});
