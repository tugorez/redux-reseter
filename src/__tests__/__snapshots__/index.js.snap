// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer should compute the operations 1`] = `2`;

exports[`redux-reseter reset creates a valid action 1`] = `
Object {
  "payload": Array [
    Object {
      "type": "ADD",
    },
    Object {
      "type": "SUB",
    },
    Object {
      "type": "SUB",
    },
  ],
  "type": "@REDUX-RESET.RESET",
}
`;

exports[`redux-reseter reseter should wait for a reset action and reset the state 1`] = `0`;
